oap-server/server-starter/src/main/resources/lal/mysql-slowsql.yaml

```diff
@@ -0,0 +1,30 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+rules:
+  - name: mysql-slowsql
+    dsl: |
+      filter {
```

Comment:
As the tag is added, you should place `if tag['isSlowSql'] == true`(pseudocode only) as precondition here.

Reply:
I set the check in the code
```
 @SuppressWarnings("unused")
    public void slowSql(@DelegatesTo(SlowSqlSpec.class) final Closure<?> cl) {
        if (BINDING.get().shouldAbort()) {
            return;
        }

        String isSlowSql = BINDING.get().log().getTags().getDataList()
                .stream()
                .filter(data -> Binding.KEY_IS_SLOW_SQL.equals(data.getKey()))
                .map(KeyStringValuePair::getValue)
                .collect(Collectors.toList()).get(0);

        if (!Boolean.parseBoolean(isSlowSql)) {
            return;
        }

        BINDING.get().databaseSlowStatement(new DatabaseSlowStatementBuilder(namingControl));

       ......
```
Need I move it to the LAL?

Reply:
Yes, all logic should be moved into LAL. SlowSQL function should only validate data, such as no service name, no statement, no layer or latency is less than 1 should not be allowed. The log should be ignored.

Reply:
I think your latest change missing `IF tag` condition here?

Reply:
`IF tag` condition is in LAL
```
 json{
        }
        extractor{
          layer parsed.layer as String
          service parsed.service as String
          timestamp parsed.time as String
          if (tag("LOG_KIND") == "SLOW_SQL") {
             slowSql {
                      id parsed.id as String
                      statement parsed.statement as String
                      latency parsed.query_time as Long
                     }
          }
        }
      }
```